### DISCUSSION TITLE
Target Audience on Hubsite not working properly

### DISCUSSION THREAD

**Discussion by kkowal89** [pnp/powershell#1269] [Q&A]
Hello dear PNP Community,

I have a strange side-effect and I am not sure if this is a issue in the pnp module or the wrong way of me implementing it.
So I try to describe my situation and context. 

What I am trying to do is setting the target audience of a link inside my hub navigation. I try to achive this with the following code:

` $url ="https://xxx.sharepoint.com/sites/SiteName"

  $title = "New Link9"

 $linkUrl = "https://XXX.de"

 $audienceId = New-Object System.Collections.Generic.List[guid]

 $audienceId.add([System.guid]::New("b62fbbef-5c3b-45ac-987f-755474006bce")) 

 Connect-PnPOnline -Url $url -Credentials $Cred

 #$ctx = Get-PnPConnection

 $context = Get-PnPContext

 $newNode = Add-PnPNavigationNode -Location TopNavigationBar -Title $title -Url $linkUrl

 $newNode.AudienceIds = $audienceId    

 $newNode.Update()   

 $context.ExecuteQuery()`

The problem is, that the code is run without any errors and the navigation seems to be extended with the added link and target audience. But when I visit the site the target audience is not working. I can still see the link even if I am not a member of the target audience. 

The interessting part is that when I press the "edit" button of the top bar na
vigation, I can see all the links and target audiences in the left bar editor, but the target audience is not applied. 

![PNPSupport1](https://user-images.githubusercontent.com/78034493/137680337-e8bffa54-633d-4339-98c2-e89d843e23fa.PNG)
When I change something in the edit panel (eg. a new link or chang
es to a link) and save it the target audience is applied and all works. But it works only when I use the SharePoint GUI and somehow its not working via code. 

Can anybody help me out with this problem?

**Comment by kkowal89** [comment]
Nothing?

**Reply by kbeeveer46** [reply]
@kkowal89  Did you ever figure this out? I'm running into the same issue. The following code works and I can see that the audience was added to my Quick Launch link when I view the site but it doesn't apply the audience. The quick launch link should be hidden from me since I'm not part of the audience but it's not hidden. When I add the audience manually to the link by clicking the edit button on the SharePoint site it works fine. But when I do it from PnP PowerShell it does not. Same exact issue you had.

```
$newNode = Add-PnPNavigationNode -Location "QuickLaunch" -Title "Audience Test" -Url "http://linkless.header/"

$audienceList = New-Object System.Collections.Generic.List[guid]
$audienceList.add([System.guid]::New(("7e5c22f2-1254-453e-84f4-58ce4ed3e995")))

$newNode.AudienceIds = $audienceList
$newNode.Update()
Invoke-PnPQuery
```
